pkg/tracing/datadog/datadog.go

```diff
@@ -44,17 +46,24 @@ func (c *Config) SetDefaults() {
 
 // Setup sets up the tracer.
 func (c *Config) Setup(serviceName string) (opentracing.Tracer, io.Closer, error) {
-	tag := strings.SplitN(c.GlobalTag, ":", 2)
+	if c.GlobalTag != "" {
+		log.WithoutContext().Warn(`Datadog: option "globalTag" is deprecated, please use "globalTags" instead.`)
 
-	value := ""
-	if len(tag) == 2 {
-		value = tag[1]
+		key, value, _ := strings.Cut(c.GlobalTag, ":")
+
+		// Don't override a tag already defined with the new option.
+		if _, ok := c.GlobalTags[key]; !ok {
+			if c.GlobalTags == nil {
+				c.GlobalTags = make(map[string]string)
+			}
+
+			c.GlobalTags[key] = value
+		}
 	}
```

Comment:
imho, it would have been better to make them mutually exclusive as soon as possible in the flow. It is highly unlikely the user purposefully uses both options at the same time, so we would do them a favor by detecting it, telling them to clean up their conf, and stopping.
But whatever, the option is going to disappear soon anyway.